pilot/pkg/model/push_context.go

```diff
@@ -307,8 +307,19 @@ type PushRequest struct {
 	// There should only be multiple reasons if the push request is the result of two distinct triggers, rather than
 	// classifying a single trigger as having multiple reasons.
 	Reason []TriggerReason
+
+	// PushType declares the type of push this is. Support values are currently PushTypeRequest, for
+	// responses to XDS requests, and PushTypePush for Istiod-originated pushes.
+	PushType PushType
 }
 
+type PushType string
```

Comment:
Can declare a string method for PushType 

func (x PushType) String () string{
     ...
}

Reply:
done

Reply:
Would it be more meaningful to tie this to TriggerReason like `Proxy Request` and log TriggerReason instead of this new type?